ocaml/xapi/xapi_vdi.ml

```diff
@@ -25,6 +25,30 @@ open Printf
 (**************************************************************************************)
 (* current/allowed operations checking                                                *)
 
+let check_sm_feature_error op sm_features sr =
```

Comment:
I think we should add a type annotation here, otherwise the compiler will infer a random polymorphic variant:
```ocaml
let check_sm_feature_error op:API.vdi_operations sm_features sr =
```
This will also enable warnings/errors in case the match is incomplete, provided we enable the relevant compiler flags.